File: 1354-construct-target-array-with-multiple-sums/1354-construct-target-array-with-multiple-sums.go
package main

func isPossible(target []int) bool {
	sum := 0
	for _, num := range target {
		sum += num
	}

	for max(target) > 1 {
		maxVal := max(target)
		restSum := sum - maxVal

		if restSum <= 0 || restSum >= maxVal {
			return false
		}

		newVal := maxVal % restSum
		if newVal == 0 && restSum > 1 {
			return false
		}

		sum = restSum + newVal
		updateMax(target, newVal)
	}
	return true
}

func max(arr []int) int {
	maxVal := arr[0]
	for _, num := range arr {
		if num > maxVal {
			maxVal = num
		}
	}
	return maxVal
}

func updateMax(arr []int, newVal int) {
	for i := range arr {
		if arr[i] == max(arr) {
			arr[i] = newVal
			break
		}
	}
}

File: 1354-construct-target-array-with-multiple-sums/1354-construct-target-array-with-multiple-sums.cpp
int init=[]{ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);return 0;}();
class Solution {
public:
    bool isPossible(std::vector<int>& target) {
        long long sum = 0;
        for (int num : target) sum += num;
        while (*std::max_element(target.begin(), target.end()) > 1) {
            int max_val = *std::max_element(target.begin(), target.end());
            long long rest_sum = sum - max_val;
            if (rest_sum <= 0 || rest_sum >= max_val) return false;
            int new_val = max_val % rest_sum;
            sum = rest_sum + new_val;
            if (new_val == 0 && rest_sum > 1) return false;
            std::replace(target.begin(), target.end(), max_val, new_val);
        }
        return true;
    }
};
